.github/workflows/shark_model_suite.yml

```diff
@@ -0,0 +1,91 @@
+# Runs the shark model suite on iree wheels from build package workflow
+name: Validate Shark Model Suite
```

Comment:
[comment here for threading]

> Is there a way to test this besides merging it and running it manually?

There *should* be, but I don't think there actually is in the UI, which seems to only give you options to run workflows that have run previously and only lets you select from branches on the main repo. It's possible that by temporarily adding a `pull_request` event here, you could get it to create a run. That will have different characteristics, like missing parameters. I think API may be the best bet. I can give it a try once some of the other stuff is cleaned up

Reply:
I apologize for the somewhat messy request. Normally I'd write some ugly code->test->cleanup. Not knowing how to test I stopped a little short at the ugly step 😅 

I suppose even if I look up how to trigger it via api its unlikely that I have the permission to run arbitrary code on google's resources, so I don't know if I can do much besides hoping it doesn't have a syntax error that I've overlooked and thanking you for your time.

Reply:
No worries, I understand the difficulty of testing these sorts of things intimately :-) What I would recommend (for testing CI things in general) is relying as little as possible on the specific CI system (so breaking run steps out into scripts), containerizing (which we already do). Of course at some point we will need to test this on the actual CI. I think for that I can help (I have various schemes in mind).

Reply:
You could also try out https://github.com/nektos/act. I haven't tried it myself though

Reply:
In order to hook this into @GMNGeoffrey's latest additions to the workflows, and with @ScottTodd's comments in mind, I want to run something that can be called by any script that includes building iree-compiler and iree-runtime from source with python bindings. Unfortunately, it looks like the existing pre/post submit workflows don't build the python bindings (or the compiler iirc). If the plan is to run the shark suite as frequently as pre/postsubmit, would it make sense to edit the existing workflow to build the remainder of the required pieces? Or would you prefer I do something standalone that builds all of shark's iree dependencies? (I was leaning towards the former to avoid building what is already built)

Reply:
Building everything, including python bindings, on GitHub actions is the plan. Just not migrated yet. Building just the runtime is deliberately its own workflow, so we wouldn't want to edit that one, but we can (and will) add one that does a full build

Reply:
I'm fine with everything being built in the same action, but not if that building is going to happen on our A100 machines, since they are low on CPU and carry a very expensive idle accelerator when compiling things. What do you actually need to run these tests? Does it need to be a full release? I'm guessing that a IREE Python build directory on the `PYTHONPATH` would be sufficient. I'm not sure how much more time it takes to package up the wheels vs just building the Python bindings, but it is at least a lot more fiddly with manylinux dockers and such.

Reply:
Your guess is right. I also agree it makes sense not to build on an a100 machine.

Reply:
There's now a job that builds all of the main IREE project and shows how to make use of the test directory. I see your edit that notes we'll need the TF tools also. I'll work on a job to build them as well.

Reply:
> There's now a job that builds all of the main IREE project and shows how to make use of the test directory. I see your edit that notes we'll need the TF tools also. I'll work on a job to build them as well.

awesome!! I'll push revisions to this shortly using those jobs!

Reply:
https://github.com/iree-org/iree/pull/9927 adds a job that builds the TF binaries and a job that uses that + the full build dir. I'm hitting some weird errors with Vulkan though

Reply:
cool thanks I'll rebase and edit this accordingly